pkg/build/generator/generator.go

```diff
@@ -532,7 +532,11 @@ func getNextBuildNameFromBuild(build *buildapi.Build) string {
 		nameElems := strings.Split(buildName, "-")
 		buildName = strings.Join(nameElems[:len(nameElems)-1], "-")
 	}
-	return fmt.Sprintf("%s-%d", buildName, int32(util.Now().Unix()))
+	suffix := fmt.Sprintf("%v", util.Now().UnixNano())
```

Comment:
what happens if this makes us exceed the max build name length? I know the old method (epoch) resulted in something at least 10 chars long, so we're no worse off, just wondering what happens